[SPARK-10669] [Docs] Link to each language's API in codetabs in ML docs: spark.mllib

keypointt opened the pull request:
In the Markdown docs for the spark.mllib Programming Guide, we have code examples with codetabs for each language. We should link to each language's API docs within the corresponding codetab, but we are inconsistent about this. For an example of what we want to do, see the "ChiSqSelector" section in https://github.com/apache/spark/blob/64743870f23bffb8d96dcc8a0181c1452782a151/docs/mllib-feature-extraction.md
This JIRA is just for spark.mllib, not spark.ml.

Please let me know if more work is needed, thanks a lot.

keypointt committed:
[SPARK-10669] test modify

keypointt committed:
[SPARK-10669] Link to each language API in codetabs in spark mllib docs

keypointt committed:
[SPARK-10669] minor correction

keypointt committed:
[SPARK-10669] re-commit, Link to each language API in codetabs in spark mllib docs

keypointt committed:
[SPARK-10669] add up API links

keypointt committed:
[SPARK-10669] undo previous wrong changes

keypointt committed:
[SPARK-10669] undo previous wrong changes for mllib-ensembles.md

keypointt committed:
[SPARK-10669] undo previous wrong changes for mllib-frequent-pattern-mining.md

AmplabJenkins commented:
Can one of the admins verify this patch?

srowen commented on docs/mllib-frequent-pattern-mining.md:
```diff
@@ -195,10 +202,12 @@ results.collect().foreach { rule =>
 
 <div data-lang="java" markdown="1">
-[AssociationRules](api/java/org/apache/spark/mllib/fpm/AssociationRules.html)
+[AssociationRules](api/scala/index.html#org.apache.spark.mllib.fpm.AssociationRules)
```
Unintentional change? I think this was correct before

keypointt replied:
I'll correct it

srowen commented on docs/mllib-linear-methods.md:
```diff
@@ -230,6 +232,8 @@ Spark Java API uses a separate `JavaRDD` class. You can convert a Java RDD to a
 calling `.rdd()` on your `JavaRDD` object. A self-contained application example
 that is equivalent to the provided example in Scala is given bellow:
 
+Refer to the [`BinaryClassificationMetrics` Java docs](api/java/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.html) for details on the API.
```
This doesn't parallel the Scala note above with SVMWithSGD?

keypointt replied:
you are right, I'll correct it

srowen commented on docs/mllib-statistics.md:
```diff
@@ -86,6 +90,8 @@ System.out.println(summary.numNonzeros()); // number of nonzeros in each column
 which contains the column-wise max, min, mean, variance, and number of nonzeros, as well as the
 total count.
 
+Refer to the [`Statistics` Python docs](api/python/pyspark.mllib.html#pyspark.mllib.stat.Statistics) for more details on the API.
```
Again should this be about MultivariateStatisticalSummary ?

keypointt replied:
I'll correct it

srowen commented:
Looks generally OK to me, with a few questions. You might also build the docs in docs/ with `SKIP_API=1 jekyll build` and spot-check your changes to make sure they look as you intended.

keypointt commented:
@srowen thank you for the code review, I'll change the code lines you mentioned above and test it out. :)

keypointt committed:
[SPARK-10669] minor changes, add markdown='1' for div

keypointt commented:
Hi @srowen I've corrected what you mentioned, and added markdown level for the divs missing markdown="1". I've run "jekyll build" and tested on my local machine too. Thanks a lot :)

SparkQA commented:
[Test build #1842 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1842/consoleFull) for   PR 8977 at commit [`5e298fb`](https://github.com/apache/spark/commit/5e298fb610ba35136cfdc6a2eeea5e9abe1b81fc).

SparkQA commented:
[Test build #1842 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1842/console) for   PR 8977 at commit [`5e298fb`](https://github.com/apache/spark/commit/5e298fb610ba35136cfdc6a2eeea5e9abe1b81fc).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
